src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
According to the dependency inversion principle, a library should not depend on a implementation (utility package), it should rely on interfaces (PSR-17).  The interoperability is PSR-17, the interfaces.

Say that package A requires U1 (utility library 1) and package B requires U2. If my app requires both A and B I would have two utility libraries doing the same thing. Where is the interoperability here? There is none... One could just skip PSR-17 completely. 

The interfaces in PSR-17 should be easy to use for a library author so no utility library is needed. 

-------
Note: There could be competing PSR-17 **implementations**. Example two different libraries to create Guzzle objects.